rasa/nlu/training_data/converters/nlg_markdown_to_yaml_converter.py

```diff
@@ -41,26 +41,26 @@ async def convert_and_write(cls, source_path: Path, output_path: Path) -> None:
         training_data = reader.read(source_path)
         converted_responses = {}
 
-        for response_name, examples in training_data.responses.items():
-            new_response_name = cls._normalize_response_name(response_name)
-            converted_responses[new_response_name] = examples
+        for utter_action, examples in training_data.responses.items():
```

Comment:
we should keep this file as it is. It's converting training data for the response selector (a NLU component). Giving it the `utter_action`  names, makes it seem as we'd be dealing with Rasa Core responses (I hope that makes sense)

Reply:
I have limited knowledge here. 😅 I don't quite understand.

Reply:
Because the Core part (at least to me) predicts and executes actions. `utter_` indicates that we are dealing with an action. In contrast the `ResponseSelector` is an NLU component which selects a response based on the user message. In the end this boils down to the fact that Rasa Open Source will utter the response which NLU selected. However, as the `ResponseSelector` selects `responses` (see docs [here](https://rasa.com/docs/rasa/reference/rasa/nlu/selectors/response_selector) or [here](https://rasa.com/docs/rasa/chitchat-faqs#2-defining-retrieval-intents-and-the-responseselector)) I'd prefer to keep the NLU vocabulary to make it clear that this isn't yet an executable action but just something which NLU predicts. Does that make sense? 🙈

Reply:
Last time in your call you explained it really well. Thanks! So now I understand. 🎊

Reply:
Actually, there is a slight problem. There is some coupling happening between `core/actions/action.py` and `nlu/response_selector.py`. https://github.com/RasaHQ/rasa/blob/master/rasa/core/actions/action.py#L400, https://github.com/RasaHQ/rasa/blob/master/rasa/nlu/selectors/response_selector.py#L433. I renamed it to `RESPONSE_SELECTOR_UTTER_ACTION_KEY`, maybe we can keep it that way? I have changed the other renamings back to response_name in nlu.

Reply:
sounds good 👍🏻